python/cudf_polars/cudf_polars/callback.py

```diff
@@ -91,9 +151,19 @@ def set_memory_resource(
     """
     if mr is None:
         device: int = gpu.getDevice()
-        mr = default_memory_resource(device)
-    previous = rmm.mr.get_current_device_resource()
-    rmm.mr.set_current_device_resource(mr)
+        rmm_mode = os.getenv("CUDF_POLARS_RMM_MODE", "managed_pool")
+        if rmm_mode == "managed_pool":
+            managed_memory_is_supported = (
+                pylibcudf.utils._is_concurrent_managed_access_supported()
+            )
+            previous = rmm.mr.get_current_device_resource()
+            mr = enable_uvm(device)
+            rmm.mr.set_current_device_resource(mr)
+            _enable_managed_prefetching(rmm_mode, managed_memory_is_supported)
+        else:
+            mr = default_memory_resource(device)
+            previous = rmm.mr.get_current_device_resource()
```

Comment:
Looks like we need to do this above the first conditional on 152 to enable the finally clause on 170 below.